SimulationAgent.Test/DeviceProperties/DevicePropertiesActorTest.cs

```diff
@@ -72,22 +88,70 @@ public void Handle_Event_Should_Throw_When_Out_Of_Range()
                 () => this.target.HandleEvent(OUT_OF_RANGE_EVENT));
         }
 
+        [Fact, Trait(Constants.TYPE, Constants.UNIT_TEST)]
+        public void Should_Return_CountOfFailedTwinUpdates_When_TwinUpdateFails()
+        {
+            // Arrange
+            const int FAILED_DEVICE_TWIN_UPDATES_COUNT = 5;
+
+            //this.SetupDevicePropertiesActor();
+            this.target.Setup(DEVICE_ID,
+                this.deviceStateActor.Object,
+                SetupDeviceConnectionActor(),
```

Comment:
why does this test need a new connection actor, instead of using the one created in the ctor? there is a lot of test setup without comments/instructions, which will make the tests hard to maintain